src/main/java/art/heredium/core/util/AuthUtil.java

```diff
@@ -0,0 +1,19 @@
+package art.heredium.core.util;
+
+import java.util.Optional;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import art.heredium.domain.account.entity.UserPrincipal;
+
+public class AuthUtil {
+  public static Optional<Long> getCurrentUserAccountId() {
+    Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+    if (!(authentication.getPrincipal() instanceof UserPrincipal)) {
+      return Optional.empty();
+    }
+    UserPrincipal userPrincipal = (UserPrincipal) authentication.getPrincipal();
+    return Optional.ofNullable(userPrincipal.getAccount().getId());
```

Comment:
```suggestion
    return Optional.of(userPrincipal.getAccount().getId());
```
Đến đây r thì account_id ko thể null dc nữa.

Reply:
Đúng vậy, nhưng còn 1 trường hợp nữa là user admin thì account sẽ null, nên tôi đã fix cả 2: cb4ac48567ccd3cfa046ec6414af3394e46a159e
https://github.com/Incuvers-IT/heredium-BE/blob/510b1d8a045a46b3aae4952dfa7dbac75802001d/src/main/java/art/heredium/domain/account/entity/UserPrincipal.java#L83-L85

Reply:
vậy thì như này cho ngắn:
```java
public static Optional<Long> getCurrentUserAccountId() {
    Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
    if (!(authentication.getPrincipal() instanceof UserPrincipal)) {
      return Optional.empty();
    }
    UserPrincipal userPrincipal = (UserPrincipal) authentication.getPrincipal();
    return Optional.ofNullable(userPrincipal.getAccount()).map(Account::getId);
  }
```

Reply:
Đồng ý: cbcb7ecf6407922420b118c631e7e50a9f56c5e7